geomesa-cassandra/geomesa-cassandra-datastore/src/test/scala/org/locationtech/geomesa/cassandra/data/CassandraDataStoreTest.scala

```diff
@@ -23,52 +22,55 @@ import org.locationtech.geomesa.features.ScalaSimpleFeature
 import org.locationtech.geomesa.index.utils.{ExplainString, Explainer}
 import org.locationtech.geomesa.utils.collection.SelfClosingIterator
-import org.locationtech.geomesa.utils.conf.GeoMesaSystemProperties.SystemProperty
 import org.locationtech.geomesa.utils.geotools.{SchemaBuilder, SimpleFeatureTypes}
-import org.locationtech.geomesa.utils.io.PathUtils
 import org.specs2.matcher.MatchResult
 import org.specs2.mutable.Specification
 import org.specs2.runner.JUnitRunner
-
-import java.nio.file.{Files, Path}
+import org.specs2.specification.BeforeAfterAll
+import org.testcontainers.cassandra.CassandraContainer
+import org.testcontainers.containers.GenericContainer
+import org.testcontainers.scylladb.ScyllaDBContainer
 
 @RunWith(classOf[JUnitRunner])
-class CassandraDataStoreTest extends Specification {
+class CassandraDataStoreTest extends Specification with BeforeAfterAll {
 
   import scala.collection.JavaConverters._
 
   sequential
 
-  var storage: Path = _
   var params: Map[String, String] = _
-  var ds: CassandraDataStore = _
 
-  step {
-    storage = Files.createTempDirectory("cassandra")
+  private val cassandraContainer: GenericContainer[?] = new CassandraContainer("cassandra:4.0.8")
+  private val scyllaContainer: GenericContainer[?] = new ScyllaDBContainer("scylladb/scylla:6.2")
+  private var dsList: List[CassandraDataStore] = List()
 
-    System.setProperty("cassandra.storagedir", storage.toString)
+  def beforeAll(): Unit = {
+    cassandraContainer.start()
+    scyllaContainer.start()
+    dsList = List(dsFromContainer(cassandraContainer), dsFromContainer(scyllaContainer))
+  }
 
-    EmbeddedCassandraServerHelper.startEmbeddedCassandra("cassandra-config.yaml", 1200000L)
+  def afterAll(): Unit = {
+    for (ds <- dsList) {
+      ds.dispose()
+    }
+    cassandraContainer.stop()
+    scyllaContainer.stop()
+  }
 
-    var readTimeout: Int = SystemProperty("cassandraReadTimeout", "12000").get.toInt
 
-    if (readTimeout < 0) {
-      readTimeout = 12000
-    }
-    val host = EmbeddedCassandraServerHelper.getHost
-    val port = EmbeddedCassandraServerHelper.getNativeTransportPort
-    val cluster = new Cluster.Builder().addContactPoints(host).withPort(port)
-        .withSocketOptions(new SocketOptions().setReadTimeoutMillis(readTimeout)).build().init()
-    val session = cluster.connect()
-    val cqlDataLoader = new CQLDataLoader(session)
+  def dsFromContainer(container: GenericContainer[?]): CassandraDataStore = {
+    val cqlSession = new Cluster.Builder().addContactPoints(container.getHost).withPort(9042)
+      .withSocketOptions(new SocketOptions().setReadTimeoutMillis(12000)).build().connect()
+    val cqlDataLoader = new CQLDataLoader(cqlSession)
     cqlDataLoader.load(new ClassPathCQLDataSet("init.cql", false, false))
 
-    params = Map(
-      Params.ContactPointParam.getName -> s"$host:$port",
+    val params = Map(
+      Params.ContactPointParam.getName -> s"${container.getHost}:9042",
```

Comment:
i think you're still going to need `container.getMappedPort(0942)` (syntax might be wrong...) here and above, otherwise both containers are going to conflict on that port.